llama-index-integrations/readers/llama-index-readers-upstage/llama_index/readers/upstage/base.py

```diff
@@ -0,0 +1,404 @@
+
+        Args:
+            files (dict): A dictionary containing the files to be sent in the request.
+
+        Returns:
+            dict: The JSON response from the API.
+
+        Raises:
+            ValueError: If there is an error in the API call.
+        """
+        try:
+            headers = {"Authorization": f"Bearer {self.api_key}"}
+            options = {"ocr": self.use_ocr}
+            response = requests.post(
+                LAYOUT_ANALYSIS_URL, headers=headers, files=files, data=options
+            )
+            response.raise_for_status()
+
+            result = response.json().get("elements", [])
+
+            return [
+                element for element in result if element["category"] not in self.exclude
+            ]
+
+        except requests.RequestException as req_err:
+            # Handle any request-related exceptions
+            print(f"Request Exception: {req_err}")
+            raise ValueError(f"Failed to send request to Upstage API: {req_err}")
+        except json.JSONDecodeError as json_err:
+            # Handle JSON decode errors
+            print(f"JSON Decode Error: {json_err}")
+            raise ValueError(f"Failed to decode JSON response: {json_err}")
+
+    def _split_and_request(
+        self,
+        full_docs: fitzDocument,
+        start_page: int,
+        num_pages: int,
+    ) -> List:
+        """
+        Splits the full pdf document into partial pages and sends a request to the
+        server.
+
+        Args:
+            full_docs (str): The full document to be split and requested.
+            start_page (int): The starting page number for splitting the document.
+            num_pages (int, optional): The number of pages to split the document
+                                       into.
+                                       Defaults to DEFAULT_NUMBER_OF_PAGE.
+
+        Returns:
+            response: The response from the server.
+        """
+        with fitz.open() as chunk_pdf:
+            chunk_pdf.insert_pdf(
+                full_docs,
+                from_page=start_page,
+                to_page=start_page + num_pages - 1,
+            )
+            pdf_bytes = chunk_pdf.write()
+
+        with io.BytesIO(pdf_bytes) as f:
+            return self._get_response({"document": f})
+
+    def _element_document(
+        self, element: Dict, output_type: OutputType, split: SplitType
+    ) -> Document:
+        """
+        Converts an elements into a Document object.
+
+        Args:
+            element (Dict): The element to be converted into a Document object.
+            output_type (OutputType): The output type of the document.
+            split (SplitType): The split type of the document.
+
+        Returns:
+            Document: A Document object representing the element with its content
+                      and metadata.
+
+        """
+        return Document(
+            text=(parse_output(element, output_type)),
+            extra_info={
```

Comment:
Shouldn't we add bbox here?

Reply:
I have added the bounding boxes you mentioned.